### PR TITLE
Also in this sidepanel

### LINKED ISSUE #8107: Create new informational side panel for hybrid learning view
**Overview**:
The new side panel component is used to display resource information and lesson plan information. 

Done when:
- Side panel can be toggled open and closed (note: this can be a last `TODO` for when the menu bar is merged)
- Side panel has the appropriate new mobile-responsive breakpoints laid out in figma
- Side panel content is rendered either as information panel or as lesson/topic panel
- As an information panel: 
      1.  The resource title, description, subject, level, language, author, license holder,  and license details are visible and displayed in line with the figma design 
      2. Long descriptions should be truncated to fit within a "preview" 170px high, and have an option to toggle to read more (with scroll) and then read less
      3. The license details should default to being collapsed, and similarly have a toggle to display that metadata 
      4. There should be a button at the bottom of the panel to download resource, and this interaction should be implemented as part of the PR 
- As a topic/lesson panel:
   1. Each related (previous/next resource) should be listed in order 
   2. The current resource should be indicated as current with a shaded background (as indicated in design)
   3. Basic metadata of the title of the activity/resources and an icon should be displayed, with tooltips
   4. Empty topics should render with a "no more topics" message (see figma references)

**Beyond scope of this ticket**:
- Creating the "chips" seen at the top of the info panel
- Side panel implemented in other place in the code base
- Interactions to navigate between topics/resources in the side panel _(as this needs further clarification)_
- Display of metadata related to progress, completion and estimated time

**NOTE**: 
- Any metadata that is within the scope of this ticket but not available should be noted and raised for discussion to clarify the timeline of when it can be included. `TODOs` should be included in the code as well to keep track, and outstanding items should be referenced in the PR.

**Figma references**:
Screen Shot 2021-05-25 at 10 36 11 AM
Screen Shot 2021-05-25 at 10 31 36 AM
Screen Shot 2021-05-25 at 10 31 28 AM
Screen Shot 2021-05-25 at 10 33 00 AM
Screen Shot 2021-05-25 at 10 32 09 AM
Screen Shot 2021-05-25 at 10 32 50 AM
Screen Shot 2021-05-25 at 10 33 52 AM
Screen Shot 2021-05-25 at 10 33 48 AM
Screen Shot 2021-05-25 at 10 34 00 AM

### LINKED ISSUE #8307: Implement Side Informational Panel throughout new Learn experience
**Overview:**
While the container component is handled in #8107, the implementation of the component, and connection to relevant cards, toolbars, etc. still needs to be completed. 

The sidepanel can be accessed 3 ways:
- from the content renderer (as a resource informational panel) 
- from the content renderer (as a topic/lesson informational panel) 
- from a single card by the (i) icon (as a resource informational panel) 

**Done when:**
- The three interactions above all work and the correct metadata is displayed for each

Screen Shot 2021-08-17 at 10 46 41 AM
Screen Shot 2021-08-17 at 10 46 56 AM
Screen Shot 2021-08-17 at 10 45 15 AM

### PR DESCRIPTION
**Pull request by nucleogenesis**
## Summary

Provides context-informed side panel for content viewing per [Figma specs](https://www.figma.com/file/uiZuCIqh2KYvBUfbCiJyyA/Hybrid-Learning?node-id=212%3A0).

Video shows, roughly in order:
- Guest user (no progresses shown - no lesson available)
- Logged in user in library context (progress shown)
- "Next folder navigation"
- Back route navigation
- Lesson content viewing
- Lesson content navigation (within the lesson)

https://user-images.githubusercontent.com/6356129/141061287-40fe4d11-7154-41dc-ada9-7b89a97ae69a.mp4

Overall, this was more complicated than I anticipated. Some notes:

- This is not particularly pretty. I added to the bloat in `LearnImmersiveLayout` with some methods that gather and map progresses to (if user is logged in) the returned nodes. It gets the Lesson model when in lesson context as well - this is all to pass it on to the `AlsoInThis` component (which is the body of the Side Panel). I hope the comments help clarify this.
- The CSS uses a lot of absolute positioning - I don't suspect it would break if the styles of the side panel width was changed but would be simple enough to fix if it did. It's not often you are given a finite space to work in the web as the side panel fits comfortably on nearly all screen sizes.
- The `detail_route` `next_content` now looks for a query param `topicOnly` - if true, then it will return the next topic. This was in order to easily get the "Next folder" content node
- ProgressBar had a `contentNode` prop, but didn't check if there was a `progress` property set - instead opting to look into a progress mapping. Perhaps I've missed something w/ that `useContentNodeProgress()` composable @rtibbles .
- I tried my hand at a quick Gherkin - definitely would love some feedback @radinamatic - I hope this covers enough and gets the idea across.
- The routing currently will go 2 resources deep - then if you back out twice it'll pop you back into the Library or Lesson you were in. This isn't intentional but I think it's a good UX overall. Interested in thoughts @jtamiace 

General thoughts on future improvements:
- `fetchSiblings` `detail_route` might be helpful and simplify this a bit? 
- Move fetching business into `AlsoInThis` to pare down `LearnImmersiveLayout` 
- `AlsoInThis` should probably be two separate components `AlsoInThisLesson` and `AlsoInThisFolder` I just wish I'd seen it earlier on :-/ 

## References

Fixes #8107 
Fixes #8307 

## Reviewer guidance

It ended up being more complicated than I had anticipated but it should be good to go :) 

Mainly be sure that you can get the 

**General**
- Progress indicator on content where there is progress. Just a star if completed
- Scrollbar appears in largely populated folders and all items are visible when scrolled down
- The content you're viewing doesn't get listed in the side bar
- Content which has a duration will show the time under the title
- A channel with lots of folders, if you view a content a level or two deep, then you should be guaranteed to see the link on the bottom of the sidebar linking to the folder.

**Lesson context**
- The contents listed should be from within the same lesson, not from siblings of that content in it's other locations (in channels in the library).
- Links should remain in Lesson context - so you should always see the "view lesson resources" icon as you click through side panel "next in lesson" resources

**Mobile**
- The progress / completed icon appears underneath the title (and optionally the duration if present)

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [x] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by jtamiace** [comment]
Minor suggestion - could you expand the titles in the 'also in this folder' to a second line?

For the back button, I think it should either _always_ go back to the previous resource (and function like the browser back), or always take them back to the lesson or folder they came from. If it does something in the middle, like the current behavior of limiting to 2 resources, we should communicate that expectation somehow, otherwise there's some amount of unpredictability of what that button will do. Since this is the main exit from the renderer, I think the back button should take them back to the lesson or folder. Otherwise, we introduce extra steps when they decide to explore that side panel, and they would have to do a workaround like opening the side panel and clicking the 'next folder' to get back faster.

**Comment by jtamiace** [comment]
An experience to follow up on in a future release is what we should do for folders with many resources. I notice that it shows the first 12 resources, which may not always be helpful for channels like GDL where the folders can be large collections of books with no real sequence.

File: kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue
Comment (marcellamaki) [inline kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue:238]: so much cleaner!

**Review by marcellamaki — COMMENTED** [review]
Overall I think this looks good, and manual QA seems good too.  

I am inclined to agree with @jtamiace that the back button should have consistent behavior. Jessica -- if you think the "always go back to the main content" (i.e. Library, Topic, lesson, etc.) is the way to go, I am happy to fold that in to my existing to-dos on the back button as a follow up issue.

The only other thing that I am a little wary of is some of the progress stuff because although I've been trying to get more familiar with it, it's something I'm still having a bit of trouble wrapping my head around. I'm going to hold off approving and merging for now. If Richard is feeling better and working tomorrow and is able to take a look, that would be great, but if not, I'm just going to review that section a second time in the morning with a fresh brain and make sure I can follow it all and I'll let you know if I have any questions. 

Thanks for your work on this, @nucleogenesis !!

**Comment by jtamiace** [comment]
> if you think the "always go back to the main content" (i.e. Library, Topic, lesson, etc.) is the way to go, I am happy to fold that in to my existing to-dos on the back button as a follow up issue.

Sounds good @marcellamaki, thank you!

**Comment by nucleogenesis** [comment]
Thanks @jtamiace - I'll need to do some style refactoring (cc @marcellamaki ) and should have this tweaked and ready to test tomorrow/early Friday.

If this needs to be merged for testing or beta-cutting then I can follow up in a separate PR as well.

File: kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue
Comment (rtibbles) [inline kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue:365]: Doesn't look like you are actually returning `Promise.all` here - maybe don't need to track the promises at all? Can just fire them off and let their changes in state do the talking.

File: kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue
Comment (rtibbles) [inline kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue:338]: No need to wait for this fetch to resolve, can use the `fetchContentNodeTreeProgress` with the same parameters: https://github.com/learningequality/kolibri/blob/release-v0.15.x/kolibri/plugins/learn/assets/src/composables/useContentNodeProgress.js#L50

Another thing to note is that the fetchTree response is paginated by default, so if it's a big topic, it won't contain everything.

File: kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue
Comment (rtibbles) [inline kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue:311]: Can filter `fetchContentNodeProgress` by `lesson_id` to avoid watiing for the two promises to resolve here. https://github.com/learningequality/kolibri/blob/release-v0.15.x/kolibri/plugins/learn/assets/src/composables/useContentNodeProgress.js#L29

Thinking we should maybe be adding the content nodes to what the LessonResource returns to avoid having to do these chained fetches (but no need to do that in this PR!).

File: kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue
Comment (rtibbles) [inline kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue:285]: It took me a few minutes to realize that this method is also setting `this.viewResourcesContents` to allow the nodes to be displayed - I see now that it's in the JSDoc comment, but a more descriptive name would help too.

**Review by rtibbles — COMMENTED** [review]
Can reduce fetches and use the composable for progress tracking.

**Review by marcellamaki — APPROVED** [review]
Looks good @nucleogenesis. I will make a follow up issue to address the progress-related suggestions that Richard included in his comments.

**Pull request merged**
